toolkit/tools/pkg/imagecustomizerlib/imagecustomizer.go

```diff
@@ -853,117 +854,130 @@ func shrinkFilesystemsHelper(buildImageFile string, verity []imagecustomizerapi.
+		if err != nil {
+			return nil, err
+		}
+		hashPartition, err := idToPartitionBlockDevicePath(verityConfig.HashDeviceId, diskPartitions, partIdToPartUuid)
+		if err != nil {
+			return nil, err
+		}
 
-	// Extract root hash using regular expressions.
-	verityOutput, _, err := shell.Execute("veritysetup", "format", dataPartition, hashPartition)
-	if err != nil {
-		return "", "", "", fmt.Errorf("failed to calculate root hash:\n%w", err)
-	}
+		// Extract root hash using regular expressions.
+		verityOutput, _, err := shell.Execute("veritysetup", "format", dataPartition, hashPartition)
+		if err != nil {
+			return nil, fmt.Errorf("failed to calculate root hash:\n%w", err)
+		}
 
-	var rootHash string
-	rootHashRegex, err := regexp.Compile(`Root hash:\s+([0-9a-fA-F]+)`)
-	if err != nil {
-		return "", "", "", fmt.Errorf("failed to compile root hash regex: %w", err)
-	}
+		rootHashRegex, err := regexp.Compile(`Root hash:\s+([0-9a-fA-F]+)`)
+		if err != nil {
+			return nil, fmt.Errorf("failed to compile root hash regex: %w", err)
+		}
 
-	rootHashMatches := rootHashRegex.FindStringSubmatch(verityOutput)
-	if len(rootHashMatches) <= 1 {
-		return "", "", "", fmt.Errorf("failed to parse root hash from veritysetup output")
-	}
-	rootHash = rootHashMatches[1]
+		rootHashMatches := rootHashRegex.FindStringSubmatch(verityOutput)
+		if len(rootHashMatches) <= 1 {
+			return nil, fmt.Errorf("failed to parse root hash from veritysetup output")
+		}
 
-	// Refresh disk partitions after running veritysetup so that the hash partition's UUID is correct.
-	diskPartitions, err = diskutils.GetDiskPartitions(loopback.DevicePath())
-	if err != nil {
-		return "", "", "", err
-	}
+		// Refresh disk partitions after running veritysetup so that the hash partition's UUID is correct.
+		diskPartitions, err = diskutils.GetDiskPartitions(loopback.DevicePath())
+		if err != nil {
+			return nil, err
+		}
 
-	// Identify the data partition UUID
-	dataPartUuid, ok := partIdToPartUuid[rootfsVerity.DataDeviceId]
-	if !ok {
-		return "", "", "", fmt.Errorf("failed to determine root partition UUID for DataDeviceId: %s", rootfsVerity.DataDeviceId)
-	}
+		// Identify the data partition UUID
+		dataPartUuid, ok := partIdToPartUuid[verityConfig.DataDeviceId]
+		if !ok {
+			return nil, fmt.Errorf("failed to determine partition UUID for DataDeviceId: %s", verityConfig.DataDeviceId)
+		}
 
-	// Identify the hash partition UUID
-	hashPartUuid, ok := partIdToPartUuid[rootfsVerity.HashDeviceId]
-	if !ok {
-		return "", "", "", fmt.Errorf("failed to find hash partition UUID for HashDeviceId: %s", rootfsVerity.HashDeviceId)
+		// Identify the hash partition UUID
+		hashPartUuid, ok := partIdToPartUuid[verityConfig.HashDeviceId]
+		if !ok {
+			return nil, fmt.Errorf("failed to find hash partition UUID for HashDeviceId: %s", verityConfig.HashDeviceId)
+		}
+
+		verityInfo[verityType] = map[string]string{
```

Comment:
I think it might be better to identify the verity devices by mount path than by verity device name.